lib/pci_caps.c

```diff
@@ -292,6 +292,44 @@ handle_px_pxdc_write(vfu_ctx_t *vfu_ctx, struct pxcap *px,
     return 0;
 }
 
+static int
+handle_px_pxlc_write(vfu_ctx_t *vfu_ctx UNUSED, struct pxcap *px,
+                     const union pxlc *const p)
+{
+    assert(px != NULL);
+    assert(p != NULL);
+
+    /* FIXME RCB only writeable for endpoints and bridges */
+    /* FIXME must preserve reserved bits */
+    px->pxlc.raw = p->raw;
+    return 0;
+}
+
+static int
+handle_px_pxsc_write(vfu_ctx_t *vfu_ctx UNUSED, struct pxcap *px,
+                     const struct pxsc *const p)
+{
+    assert(px != NULL);
+    assert(p != NULL);
+
+    /* FIXME */
+    px->pxsc.stuff = p->stuff;
+    return 0;
+}
+
+static int
+handle_px_pxrc_write(vfu_ctx_t *vfu_ctx UNUSED, struct pxcap *px,
+                     const struct pxrc *const p)
+{
+    assert(px != NULL);
+    assert(p != NULL);
+
+    /* FIXME  */
+    /* FIXME must preserve reserved bits */
+    px->pxrc.stuff = p->stuff;
```

Comment:
this matters a bit less I suppose, as it's root port specific.

Reply:
ack